### DISCUSSION TITLE
What SSAO algorithm does Filament use?

### DISCUSSION THREAD

**Discussion by Nolram12345** [google/filament#7797] [General]
I'm curious about what SSAO algorithm Filament uses. From what I can tell, part of it is SAO (Scalable Ambient Obscurance), but there's also bits of GTAO it would seem?

**Comment by romainguy** [comment]
Filament uses SAO. We don't use GTAO, just a technique from GTAO for the specific case of approximating multi-bounce AO ("colored" AO).

**Reply by Nolram12345** [reply]
Ah, fascinating! I'm curious why SAO was picked over other methods, say a horizon-based approach (which has also shown itself successful on mobile)? 
And have considerations for moving more of it to a GTAO-like core been made? I believe Sebastian Aaltonen has managed to get GTAO working at full framerate even on lower end mobile devices recently. But I understand that it is obviously not a big priority since the current algorithm is mostly adequate 👍

**Reply by romainguy** [reply]
Based on our tests it was the best performance/quality compromise.
